presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/AddIntermediateAggregations.java

```diff
@@ -120,7 +120,7 @@ public boolean isEnabled(Session session)
             source = ExchangeNode.gatheringExchange(idAllocator.getNextId(), ExchangeNode.Scope.LOCAL, source);
         }
 
-        return Optional.of(aggregation.replaceChildren(ImmutableList.of(source)));
+        return Result.replace(aggregation.replaceChildren(ImmutableList.of(source)));
```

Comment:
`Result.replace` looks a bit like "replace result" while it means "result that replaces". 

Maybe `Result.replacement`?